src/DiskIO/IpcIo/IpcIoFile.cc

```diff
@@ -372,6 +372,8 @@ IpcIoFile::push(IpcIoPendingRequest *const pending)
                dbName << " overflow: " <<
                SipcIo(KidIdentifier, ipcIo, diskId)); // TODO: report queue len
         // TODO: grow queue size
+        if (!pending->readRequest && ipcIo.page)
```

Comment:
Please either do not check `pending->readRequest` or explain why this check is necessary. Please note that I know that the current code does not pre-allocate ipcIo.page for reading requests, but that fact does not matter here.